### PR TITLE
Add Random Battles chat plugin

### PR DESCRIPTION
**Pull request by KrisXV**
Im aware the randomSets are extremely counterintuitive, im going to work on that after this

### CODE REVIEW

**Comment by KrisXV** [comment]
randomdoublesbattle doesnt work for gens outside of 8 at the moment so im looking into that

File: server/chat-plugins/random-battles.ts [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+			return this.sendReplyBox(`<span style="color:#999999;">Moves for ${template.species} in ${formatName}:</span><br />${gscMoves}`);
+		}
+		if (!template.randomBattleMoves) {
+			return this.errorReply(`Error: No moves data found for ${template.species}${`gen${dex.gen}` in GEN_NAMES ? ` in ${GEN_NAMES[`gen${dex.gen}`]}` : ``}.`);
+		}
+		const moves = template.randomBattleMoves.map(formatMove);
+		this.sendReplyBox(`<span style="color:#999999;">Moves for ${template.species} in ${formatName}:</span><br />${moves.join(`, `)}`);
+	},
+	randombattleshelp: [
+		`/randombattles OR /randbats [pokemon], [gen] - Displays a Pok\u00e9mon's Random Battle Moves. Defaults to Gen 8.`,
```
Comment (Zarel) [inline server/chat-plugins/random-battles.ts]: Shouldn't it default to the current battle's gen in a battle room?

Comment (KrisXV) [reply server/chat-plugins/random-battles.ts]: Good point; will add soon

Comment (KrisXV) [reply server/chat-plugins/random-battles.ts]: did

**Pull request merged**